.github/workflows/dependabot-auto-merge.yml

```diff
@@ -18,12 +18,22 @@ jobs:
           check-name: test
           repo-token: ${{ secrets.GITHUB_TOKEN }}
           wait-interval: 10
+      - name: Approve the PR
+        if: ${{contains(env.AUTO_MERGE, steps.metadata.outputs.dependency-names) && contains(env.SEMVER, steps.metadata.outputs.update-type)}}
+        uses: hmarr/auto-approve-action@v3.1.0
+  dependabot:
+    needs: auto_approval
+    name: Auto merge qualifying Dependabot PR
+    runs-on: ubuntu-latest
+    if: ${{ github.actor == 'dependabot[bot]' }}
+    steps:
       - name: Dependabot metadata
         id: metadata
         uses: dependabot/fetch-metadata@v1.3.4
         with:
           github-token: "${{ secrets.GITHUB_TOKEN }}"
       - name: Enable auto-merge for Dependabot PRs
+        # TODO: figure out how to de-duplicate the conditional below
```

Comment:
Was hoping to define a separate job (e.g. `check_qualifies`) with its own `if` that consolidates everything, and then make both the `auto_approval` and `dependabot` jobs have a `needs` dependency on the `check_qualifies` job.

Had a play with that approach [here](https://github.com/ChrisBAshton/test-auto-merge/pull/4) but didn't have much luck, so this feels like an optimisation we can make later.